Early stopping and config saving

JulianKlug opened the pull request.

JulianKlug committed:
Early stopping and config saving

JulianKlug commented on utils/utils.py:
```diff
@@ -220,14 +220,18 @@ def append_df_to_excel(filename, df, sheet_name='Sheet1', startrow=None,
     # get the model path with the epoch of the best model
+    # Todo - model path is not yet accurate
     model_path = os.path.join(model.save_dir, '{0:03d}_net_{1}.pth'.format(
-        val_loss_log.loc[val_loss_log['Seg_Loss'] == best_loss, 'epoch'].item(),
+        model.best_epoch,
         json_opts.model.model_type))
```
@Jimmy2027 : How do you name models upon saving and when do you save them to find them like this here?

Jimmy2027 replied:
my save function:
```
def save(self, network_label, epoch_label):
       self.save_network(self.net, network_label, epoch_label, self.gpu_ids)
```
I save the weights of the model every epoch, it takes about 0.07077169418334961 secs.

JulianKlug committed:
Save model on every improving epoch
- time overhead seems to be small < 0.1s
- remove older model saves
- better saved model naming

Jimmy2027 reviewed (commented):
looks good
see https://github.com/MonsieurWave/PerfusionCT-Net/pull/14 for another way to overwrite old checkpoints

JulianKlug committed:
Merge branch 'master' into early_stopping
